api/settings/v1/recording_rules.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package settings.v1;
+
+import "types/v1/types.proto";
+
+service RecordingRulesService {
```

Comment:
In order to be consistent with CollectionRulesService, I would consider:

- Standardize either on `Upsert` or `Insert`. I assume updating of rules should be possilbe?
- Also provide a `Get`

Reply:
I think Upsert is good. I will change to that.

I added a `Get` API in https://github.com/grafana/pyroscope/pull/3946/commits/951fd890cd8fcc8301a15adc6460bda5c029452c.

Reply:
Switched to upsert in https://github.com/grafana/pyroscope/pull/3946/commits/dfbb10beb7c0fc54d1535b347ea6ee8e8c2e5591.